lib/src/clojure_lsp/feature/diagnostics.clj

```diff
@@ -185,47 +184,64 @@
                          (q/nses-and-dependents-analysis project-db var-nses))
         var-used? (fn [var-def]
                     (some var-usages (q/var-definition-signatures var-def)))
-        kw-definitions (remove (partial exclude-public-diagnostic-definition? config) kw-defs)
-        ;; TODO: skip if no kw-definitions
-        kw-usages (into #{}
-                        (comp
-                          q/xf-all-keyword-usages
-                          (map q/kw-signature))
-                        (:analysis project-db))
+        kw-definitions (remove (partial exclude-public-diagnostic-definition? kondo-config) kw-defs)
+        kw-usages (if (seq kw-definitions) ;; avoid looking up thousands of keyword usages if these files don't define any keywords
```

Comment:
Nice improvement

Reply:
Yeah, this was a simple fix that should help a lot.